lib/realtime/api.ex

```diff
@@ -133,6 +137,16 @@ defmodule Realtime.Api do
     |> repo_replica.preload(:extensions)
   end
 
+  def get_cached_tenant(external_id) do
+    {status, value} =
+      Cachex.fetch(:db_cache, external_id, fn ->
+        get_tenant_by_external_id(external_id)
```

Comment:
@abc3 we have to check the result of this db operation and then conditionally `:commit` or `:ignore`. Fallback function should return a tuple. Let's return either `{:commit, %Tenant{}}` or `{:ignore, nil}`.

Also, we don't have to set this fallback function or expire (https://github.com/supabase/realtime/pull/328/files#diff-5adc67952ec1b4291e4fffde5db81f2b734aa61ba0e416cb244c953e54c2b25bR146) every time. We can just do it once on `start_link`.

See https://github.com/whitfin/cachex/blob/c3bf7d9cacd47db8cf7dae56a3a2c290b20d7218/lib/cachex.ex#L188 and https://github.com/whitfin/cachex/blob/c3bf7d9cacd47db8cf7dae56a3a2c290b20d7218/lib/cachex.ex#L163.

We should also think about adding in a cache limit too so we don't OOM: https://github.com/whitfin/cachex/blob/c3bf7d9cacd47db8cf7dae56a3a2c290b20d7218/lib/cachex.ex#L234

Reply:
@chasers did you set a cachex limit for logflare?

Reply:
this function is proxying before `get_tenant_by_external_id`, it can return `Tenant%{}` or `nil`

Reply:
> If the Tuple is tagged :commit the value will be placed into the cache and then returned. If tagged :ignore the value will be returned without being written to the cache. If you return a value which does not fit this structure, it will be assumed that you are committing the value.

- https://hexdocs.pm/cachex/Cachex.html#fetch/4

Right but why are you trying to commit `nil` and then setting the new expiration. Just unnecessary work.

Reply:
> If you return a value which does not fit this structure, it will be assumed that you are committing the value.

ohh, I missed that 🤦‍♂️

Reply:
> Also, we don't have to set this fallback function or expire every time. We can just do it once on start_link.

I would like not to link the cache to the current process and to have a standalone cache service tree

Reply:
> standalone cache service tree

Right, so you can start_link in `Realtime.Application.start/2` no?